File: apps/macos/Sources/ClaudeBar/API/UsageTypes.swift
import Foundation

enum UsageStatus: String, Sendable {
    case ok
    case offline
    case rateLimited = "rate-limited"
    case unauthenticated
}

struct UsageBucket: Sendable, Equatable {
    var percent: Double
    var resetsAt: Date?
}

struct UsageSnapshot: Sendable, Equatable {
    var session: UsageBucket
    var weekly: UsageBucket
    var status: UsageStatus
    var fetchedAt: Date

    static func empty(status: UsageStatus) -> UsageSnapshot {
        UsageSnapshot(
            session: UsageBucket(percent: 0, resetsAt: nil),
            weekly: UsageBucket(percent: 0, resetsAt: nil),
            status: status,
            fetchedAt: Date()
        )
    }
}

File: apps/windows/Api/UsageTypes.cs
namespace ClaudeBar.Api;

public enum UsageStatus
{
    Ok,
    Offline,
    RateLimited,
    Unauthenticated,
}

public sealed record UsageBucket(double Percent, DateTimeOffset? ResetsAt);

public sealed record UsageSnapshot(
    UsageBucket Session,
    UsageBucket Weekly,
    UsageStatus Status,
    DateTimeOffset FetchedAt)
{
    public static UsageSnapshot Empty(UsageStatus status) => new(
        new UsageBucket(0, null),
        new UsageBucket(0, null),
        status,
        DateTimeOffset.UtcNow);
}
